play-client/src/main/scala/endpoints/EndpointPlayClient.scala

```diff
@@ -28,21 +28,20 @@ class EndpointPlayClient(wsClient: WSClient)(implicit ec: ExecutionContext) exte
   /**
     * A function that, given an `A` information and a `WSRequest`, eventually returns a `WSResponse`
     */
-  type RequestEntity[A] = (A, WSRequest) => Future[WSResponse]
+  type RequestEntity[A] = (A, WSRequest) => WSRequest
+
+  override val emptyRequestEntity: RequestEntity[Unit] = { case (_, req) => req }
 
-  def get[A, B](url: Url[A], headers: RequestHeaders[B])(implicit tupler: Tupler[A, B]): Request[tupler.Out] =
-    (ab: tupler.Out) => {
-      val (a, b) = tupler.unapply(ab)
-      val wsRequest = wsClient.url(url.encode(a))
-      headers(b, wsRequest).get()
-    }
 
-  def post[A, B, C, AB](url: Url[A], entity: RequestEntity[B], headers: RequestHeaders[C])(implicit tuplerAB: Tupler.Aux[A, B, AB], tuplerABC: Tupler[AB, C]): Request[tuplerABC.Out] =
+  override def request[A, B, C, AB](
```

Comment:
> In normal code when you're using and implementing a thing at the same moment it's ok. But when you can implement something in module X and end up with error few days later in module Y it is not so nice.

Yes, I totally understand, and that’s why I think it would be useful to have an `@implement` annotation that would give you the error directly in `X`.

> Just curious, could you tell more about such case?

It can be simply because at the beginning of your project class `B` implements a `foo` abstract method of a class `A`, but then after some time this `foo` method is finally implemented directly in `A`. If you used `override` in `B` then you will not notice that there is no need anymore to override the method.

You can also have issues in more complex cases like this one:

~~~ scala
trait Base { def foo: Int }
trait Impl1 extends Base { def foo = 1 }
trait Impl2 extends Base { def foo = 2 }
object Impl extends Base with Impl1 with Impl2 // Does not compile: you have to disambiguate which `foo` you want to use
~~~

But:

~~~ scala
trait Base { def foo: Int }
trait Impl1 extends Base { override def foo = 1 }
trait Impl2 extends Base { override def foo = 2 }
object Impl extends Base with Impl1 with Impl2 // Compiles fine: Impl2 wins
~~~